pkg/client/linstor.go

```diff
@@ -372,6 +378,101 @@ func (s *Linstor) Create(ctx context.Context, vol *volume.Info, params *volume.P
 	return nil
 }
 
+func (s *Linstor) Clone(ctx context.Context, vol, src *volume.Info, params *volume.Parameters, topologies *csi.TopologyRequirement) error {
+	logger := s.log.WithFields(logrus.Fields{
+		"source": fmt.Sprintf("%+v", src),
+		"volume": fmt.Sprintf("%+v", vol),
+	})
+
+	logger.Debug("reconcile resource group from storage class")
+
+	rGroup, err := s.reconcileResourceGroup(ctx, params)
+	if err != nil {
+		logger.Debugf("reconcile resource group failed: %v", err)
+		return err
+	}
+
+	logger.Debugf("check for clone status")
+
+	status, err := s.client.ResourceDefinitions.CloneStatus(ctx, src.ID, vol.ID)
+	if err != nil && !errors.Is(err, lapi.NotFoundError) {
+		logger.Debugf("clone status failed: %v", err)
+		return err
+	}
+
+	if errors.Is(err, lapi.NotFoundError) {
```

Comment:
this is a bit weird and "jumpy". maybe do it like

```go
if err != nil {
  if errors.Is(err, lapi.NotFoundError) {
    ...
  } else {
    logger...
    return err
  }
}
```

or the other way around to have the 2 liner in the true part. something like that